app/utils/wellknown.js

```diff
@@ -0,0 +1,116 @@
+// thanks to npm well-known-wallets-hns 
+import https from "https";
+import hdns from "hdns";
+
+const init = hsd => hdns.setServers(['103.196.38.38', '103.196.38.39', '103.196.38.40'] );
+
```

Comment:
These servers do not have public keys prefixed and so they are like using http instead of https - any data retrieved from them is unsigned and therefore not secure. Notice in the [readme for hdns](https://github.com/handshake-org/hdns) that the examples have public keys included with the IP addresses, this is for SIG0 verification.

[letsdane](https://github.com/buffrr/letsdane#lets-dane-with-handshake) also uses the public key : ip construction to ensure that the data is authentic.

We might want to check that the hard-coded servers in hdns are still working (or update them if they are not): https://github.com/handshake-org/hdns/blob/276351e08f95677461c6fa8cc78911101f97aa99/lib/resolvconf.js#L10-L14

Reply:
Is it better use doh query tlsa record. If so , https://query.hdns.io/dns-query maybe better. It's provided by hdns.io.

Reply:
Yes HTTPS would be more secure but ironically that relies on legacy DNS and CA. So I think that's probably fine for convenience users but for cypherpunks itd be nice to have the option to plug in a full `pubkey@IP` identity for a trusted HNS resolver, or even connect to Fingertip by allowing the user to set proxy information like Firefox. This feature could also check if the user has configured Bob's DNS servers to be active and might be able to make a recursive request to itself. @rithvikvibhu can may chime in on this -- by default the hsd recursive port is 5350...

My last suggestion would be to use a stub resolver from bns to recursively resolve the TLSA record, anchored by the NS/DS record pair pulled directly from the blockchain. This would alleviate the need for Bob's ns and rs servers to be active.